src/sagemaker/model.py

```diff
@@ -364,12 +367,49 @@ def register(
         model_package = self.sagemaker_session.create_model_package_from_containers(
             **model_pkg_args
         )
+        if isinstance(self.sagemaker_session, PipelineSession):
```

Comment:
we handled the case where the model data is a pipeline variable when preparing container def right?

Reply:
Yes, the `model_data` type is the key to determine if we need compile time or runtime repack.
See code below in `Model$_upload_code()`